live-build/includes.chroot/usr/local/bin/flash-image.sh

```diff
@@ -156,27 +164,121 @@ function part_select() {
     done
 }
 
-# This is to evade any race conditions with the display buffer that cuts off
-# text. See votingworks/vx-iso#21.
-sleep 1
-clear
-## Detect if this disk already has VotingWorks data on it and copy the machine config
-#vg=$(vgscan | sed -s 's/.*"\(.*\)".*/\1/g')
-#
-#if [[ -n $vg ]]; then
-#
-#    dir=$(find "/dev/$vg" -name "var")
-#    mount "$dir" /mnt
-#
-#    if [ -d "/mnt/vx/config" ]; then
-#        echo "Found /vx/config to copy. Press Return to continue."
-#        read -r
-#        tar -czvf vx-config.tar.gz /mnt/vx/config
-#    fi
-#
-#    umount /mnt
-#fi
-#clear
+function detect_existing_vx_config() {
+
+  vg=$(vgscan | sed -s 's/.*"\(.*\)".*/\1/g')
+
+  if [[ -n $vg && $vg == "Vx-vg" ]]; then
+
+    vx_config_mnt="/tmp/vx_config_mnt"
+    vx_root_mnt="/tmp/vx_root_mnt"
+    mkdir -p $vx_config_mnt
+    mkdir -p $vx_root_mnt
+
+    # Activate the volume group
+    vgchange -ay $vg
+    
+    # Find the encrypted volume
+    dir=$(find "/dev/$vg" -name "var_encrypted")
+
+    # Open the encrypted volume
+    (echo "" | cryptsetup open $dir var_decrypted) || (echo "insecure" | cryptsetup open $dir var_decrypted)
+
+    mount /dev/mapper/var_decrypted $vx_config_mnt
+    mount /dev/mapper/Vx--vg-root $vx_root_mnt
+
+    if ! grep 'luks,tpm2-device=auto' ${vx_root_mnt}/etc/crypttab > /dev/null; then
+      previous_secure_boot_state=0
+    else
+      previous_secure_boot_state=1
+    fi
+
+    if [ -d "${vx_config_mnt}/vx/config" ]; then
+      previous_machine_type=$(cat ${vx_config_mnt}/vx/config/machine-type)
+      previous_machine_id=$(cat ${vx_config_mnt}/vx/config/machine-id)
+      tar --exclude="${vx_config_mnt}/vx/config/app-flags" -czvf vx-config.tar.gz ${vx_config_mnt}/vx/config
```

Comment:
does this create the tar backup in the current working directory? Does that make it depend on where this script is run from? If so, should we be more consistent about where this goes?

Reply:
@benadida, yes, it is in the current working directory. Since this is all handled by vx-iso functionality, we can be confident it's where we expect it to be, but as we work to make this script a bit less intimidating, I like the idea of defining the full path.